src/internal/metric_factory.cc

```diff
@@ -0,0 +1,149 @@
+#include "broker/internal/metric_factory.hh"
+
+#include <caf/actor_system.hpp>
+
+namespace broker::internal {
+
+// -- 'imports' to safe ourselves some typing ----------------------------------
+
+using dbl_gauge = metric_factory::dbl_gauge;
+
+using dbl_gauge_family = metric_factory::dbl_gauge_family;
+
+using int_gauge = metric_factory::int_gauge;
+
+using int_gauge_family = metric_factory::int_gauge_family;
+
+using dbl_counter = metric_factory::dbl_counter;
+
+using dbl_counter_family = metric_factory::dbl_counter_family;
+
+using int_counter = metric_factory::int_counter;
+
+using int_counter_family = metric_factory::int_counter_family;
+
+using dbl_histogram = metric_factory::dbl_histogram;
+
+using dbl_histogram_family = metric_factory::dbl_histogram_family;
+
+using int_histogram = metric_factory::int_histogram;
+
+using int_histogram_family = metric_factory::int_histogram_family;
+
+// -- core metrics -------------------------------------------------------------
+
+using core_t = metric_factory::core_t;
+
+int_gauge_family* core_t::connections_family() {
+  return reg_->gauge_family("broker", "connections", {"type"},
+                            "Number of active network connections.");
+}
+
+core_t::connections_t core_t::connections_instances() {
+  auto fm = connections_family();
+  return {
+    fm->get_or_add({{"type", "native"}}),
+    fm->get_or_add({{"type", "web-socket"}}),
+  };
+}
+
+int_counter_family* core_t::processed_messages_family() {
+  return reg_->counter_family("broker", "processed-messages", {"type"},
+                              "Total number of processed messages.", "1", true);
```

Comment:
Note: this marks the `counter` as "total value" (passing `true` at the end) to have the exporter add a `_total` suffix for Prometheus as requested in https://github.com/zeek/broker/issues/250 (`promtool` findings).

Reply:
Thanks.